src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -685,12 +677,12 @@ int uct_rc_mlx5_iface_is_reachable(const uct_iface_h tl_iface,
                                    const uct_device_addr_t *dev_addr,
                                    const uct_iface_addr_t *iface_addr)
 {
-    const uct_rc_mlx5_iface_addr_t *addr = (uct_rc_mlx5_iface_addr_t*)iface_addr;
-    uct_rc_mlx5_iface_common_t *iface    = ucs_derived_of(tl_iface,
-                                                          uct_rc_mlx5_iface_common_t);
+    const uint8_t *addr               = (uint8_t*)iface_addr;
+    uct_rc_mlx5_iface_common_t *iface = ucs_derived_of(tl_iface,
+                                                       uct_rc_mlx5_iface_common_t);
     if ((addr != NULL) &&
         (!UCT_RC_MLX5_TM_ENABLED(iface) !=
-         !(addr->flags & UCT_RC_MLX5_IFACE_ADDR_FLAG_TM))) {
+         !(*addr & UCT_RC_MLX5_IFACE_ADDR_FLAG_TM))) {
```

Comment:
revert to be a comparison rather than bit checking, because it is not a flag

Reply:
reverted